lib/charms/opensearch/v0/opensearch_tls.py

```diff
@@ -382,3 +412,156 @@ def get_unit_certificates(self) -> Dict[CertType, str]:
                 certs[CertType.APP_ADMIN] = admin_secrets["cert"]
 
         return certs
+
+    def _create_keystore_pwd_if_not_exists(self, scope: Scope, alias: str):
+        """Create passwords for the key stores if not already created."""
+        keystore_pwd = self.charm.secrets.get(scope, f"keystore-password-{alias}")
+        if not keystore_pwd:
+            self.charm.secrets.put(scope, f"keystore-password-{alias}", generate_password())
+
+    def store_new_ca(self, secrets: Dict[str, Any]):
+        """Add new CA cert to trust store."""
+        store_pwd = self.charm.secrets.get(Scope.APP, "keystore-password-ca")
+
+        keytool = f"sudo {self.jdk_path}/bin/keytool"
+
+        if not (secrets.get("ca-cert") and store_pwd):
+            logging.error("CA cert or keystore password not found, quitting.")
+            return
+
+        alias = "ca"
+        store_path = f"{self.certs_path}/{alias}.p12"
+
+        with tempfile.NamedTemporaryFile(mode="w+t") as ca_tmp_file:
+            ca_tmp_file.write(secrets.get("ca-cert"))
+            ca_tmp_file.flush()
+
+            run_cmd(
```

Comment:
@reneradoi we have a module named `opensearch_keystore`, which has a `Keystore` class that already manages the keytool and runs these types of commands. OpenSearchTLS should also inherit from `Keystore` class. This way, we have one single point of touch for `keytool` commands.

We can do on a following PR thou, can you open a ticket for this?